### PR TITLE
saas-11.5: make related fields readonly by default

### PR DESCRIPTION
**Pull request by Elkasitu**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Great initiative! Then I suppose that the side effect of always saving related fields without changing the value is still not fixed at framework level, isn't it? At least with this, by default we don't have that nasty side effect (I have reported/fixed myself several cases on core).

File: addons/l10n_ch/models/account_invoice.py [hunk 1]
```diff
@@ -26,7 +26,7 @@ class AccountInvoice(models.Model):
     l10n_ch_isr_valid = fields.Boolean(compute='_compute_l10n_ch_isr_valid', help='Boolean value. True iff all the data required to generate the ISR are present')
 
     l10n_ch_isr_sent = fields.Boolean(defaut=False, help="Boolean value telling whether or not the ISR corresponding to this invoice has already been printed or sent by mail.")
-    l10n_ch_currency_name = fields.Char(related='currency_id.name', string="Currency Name", help="The name of this invoice's currency") #This field is used in the "invisible" condition field of the 'Print ISR' button.
+    l10n_ch_currency_name = fields.Char(related='currency_id.name', string="Currency Name", help="The name of this invoice's currency") #This field is used in the "invisible" condition field of the 'Print ISR' button. readonly=False,
```
Comment (Elkasitu) [inline addons/l10n_ch/models/account_invoice.py]: defeated by a comment

File: addons/mail/models/res_users.py [hunk 2]
```diff
@@ -24,7 +24,7 @@ class Users(models.Model):
     alias_contact = fields.Selection([
         ('everyone', 'Everyone'),
         ('partners', 'Authenticated Partners'),
-        ('followers', 'Followers only')], string='Alias Contact Security', related='alias_id.alias_contact')
```
Comment (Elkasitu) [inline addons/mail/models/res_users.py]: wot in tarnation

File: addons/mrp/models/stock_move.py [hunk 3]
```diff
@@ -17,7 +17,7 @@ class StockMoveLine(models.Model):
         'Quantity Finished Product', digits=dp.get_precision('Product Unit of Measure'),
         help="Informative, not used in matching")
     done_wo = fields.Boolean('Done for Work Order', default=True, help="Technical Field which is False when temporarily filled in in work order")  # TDE FIXME: naming
-    done_move = fields.Boolean('Move Done', related='move_id.is_done', store=True)  # TDE FIXME: naming
+    done_move = fields.Boolean('Move Done', related='move_id.is_done', store=True)  # TDE FIXME: naming readonly=False,
```
Comment (Elkasitu) [inline addons/mrp/models/stock_move.py]: doot

File: addons/website_sale/models/sale_order.py [hunk 4]
```diff
@@ -250,4 +250,4 @@ def _compute_name_short(self):
             record.name_short = record.product_id.with_context(display_default_code=False).display_name
 
     def get_description_following_lines(self):
-        return self.name.splitlines()[1:]
\ No newline at end of file
+        return self.name.splitlines()[1:]
```
Comment (Elkasitu) [inline addons/website_sale/models/sale_order.py]: ???

File: odoo/addons/base/models/res_users.py [hunk 5]
```diff
@@ -248,8 +248,8 @@ def _get_company(self):
 
     # overridden inherited fields to bypass access rights, in case you have
     # access to the user but not its corresponding partner
-    name = fields.Char(related='partner_id.name', inherited=True)
-    email = fields.Char(related='partner_id.email', inherited=True)
+    name = fields.Char(related='partner_id.name', inherited=True, readonly=False)
+    email = fields.Char(related='partner_id.email', inherited=True, readonly=False)
```
Comment (Elkasitu) [inline odoo/addons/base/models/res_users.py]: clash ?

File: addons/account/models/account_move.py [hunk 6]
```diff
@@ -623,13 +623,13 @@ def _get_counterpart(self):
...
         help='Credit journal items that are matched with this journal item.')
-    journal_id = fields.Many2one('account.journal', related='move_id.journal_id', string='Journal',
+    journal_id = fields.Many2one('account.journal', related='move_id.journal_id', string='Journal', readonly=False,
         index=True, store=True, copy=False)  # related is required
     blocked = fields.Boolean(string='No Follow-up', default=False,
         help="You can check this box to mark this journal item as a litigation with the associated partner")
     date_maturity = fields.Date(string='Due date', index=True, required=True,
         help="This field is used for payable and receivable journal entries. You can put the limit date for the payment of this line.")
-    date = fields.Date(related='move_id.date', string='Date', index=True, store=True, copy=False)  # related is required
+    date = fields.Date(related='move_id.date', string='Date', index=True, store=True, copy=False)  # related is required readonly=False,
```
[the remark this answers is not in the document]
Comment (Elkasitu) [reply addons/account/models/account_move.py]: I did, thanks!

File: odoo/addons/base/models/res_users.py (hunk 5 above)
Comment (rco-odoo) [reply odoo/addons/base/models/res_users.py]: Nope. The ORM will not define those fields as inherited, because they are defined on the child model.

**Pull request merged**